farm/evaluation/metrics.py

```diff
@@ -189,11 +206,41 @@ def squad_f1_single(pred, label, pred_idx=0):
     f1 = (2 * precision * recall) / (precision + recall)
     return f1
 
+
+def confidence(preds):
+    conf = 0
+    for pred in preds:
+        conf += pred[0][0].score
+    return conf/len(preds) if len(preds) else 0
+
+
+def metrics_per_bin(preds, labels):
+    pred_bins = [[] for _ in range(10)]
+    label_bins = [[] for _ in range(10)]
+    count_per_bin = [0]*10
+    for (pred, label) in zip(preds, labels):
+        current_score = pred[0][0].score
+        if current_score == 1.0:
+            current_score = 0.9999
+        pred_bins[int(current_score*10)].append(pred)
+        label_bins[int(current_score*10)].append(label)
+        count_per_bin[int(current_score*10)] += 1
+
+    em_per_bin = [0]*10
+    confidence_per_bin = [0]*10
+    for i in range(1, 10):
```

Comment:
Not sure are we intended to skip `index=0`?

Reply:
You are right. It should be `range(0, 10)`

Reply:
or better range(num_bins)